mixed-content/generic/common.js

```diff
@@ -0,0 +1,238 @@
+function normalizePort(targetPort) {
+  var defaultPorts = [80, 443];
+  var isDefaultPortForProtocol = (defaultPorts.indexOf(targetPort) >= 0);
+
+  return (targetPort == "" || isDefaultPortForProtocol) ?
+          "" : ":" + targetPort;
+}
+
+function setAttributes(el, attrs) {
+  attrs = attrs || {}
+  for (var attr in attrs)
+    el[attr] = attrs[attr];
+}
+
+function createElement(tagName, attrs, parent) {
+  var el = document.createElement(tagName);
+  setAttributes(el, attrs);
+  if (parent && parent.appendChild)
```

Comment:
Well, every time I specify a parent for the element that gets created. For example a source tag as child of a picture tag.